### DISCUSSION TITLE
Anvil, cannot get the NFT in metamask

### DISCUSSION THREAD

**Discussion by r0bchain** [Cyfrin/foundry-full-course-cu#2041] [General]
Hi there,
I was able to deploy to Sepolia and get the NFT in Metask without issues, but I cannot say the same with Anvil, this is what I got.

I followed these steps:

1. The contract ID is correct (0xe7f1725e7734ce288f8367e1bb143e90bb3f0512), I got it from the console after run the command:
` forge test --match-test testViewTokenUri -vvvv 

![Screenshot from 2024-07-17 11-38-12](https://github.com/user-attachments/assets/cd5a9947-e82d-48e5-baf2-a7f49273dacb)

2. I use an Anvil account instead of a random PRANK user, I wanna make sure that the NFT is minted to an existing account in Anvil.
The NFT was minted to the Anvil account "**0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266**"
![Screenshot from 2024-07-17 11-40-50](https://github.com/user-attachments/assets/1591b14f-42ac-47d3-a51e-4a1c8022e81a)

![Screenshot from 2024-07-17 11-36-57](https://github.com/user-attachments/assets/6373af91-4ea7-4dd8-aefa-a44044c890ae)

4. Token ID is cero, this is also correct.

But when I try to add the NFT in Metamask, I get this error:

![AnvilNFT](https://github.com/user-attachments/assets/4d141162-e909-4408-8c3c-3834bb97fd0e)

Note: I also double-checked that Metamask is in the local Anvil network.

`

**Comment by EngrPips** [comment]
Hello @r0bertinski, Anvil might be having issues fetching the NFT, but honestly, I wouldn't bother too much since it worked perfectly on Sepolia. I can try to see what the issue is on my end and update you with whatever I find.

**Reply by r0bchain** [reply]
Yeah, thanks @EngrPips , I am not very worried about this issue, I just wanna report it, maybe it can help others with the same issue.

**Reply by EngrPips** [reply]
Yeah, it's great that you mentioned it. I will look into it and see if I find something.

**Reply by ptn77** [reply]
I'm also running into the same issue. Any update why we can't add nft from Anvil?

**Reply by EngrPips** [reply]
I am unsure, but if you can add it to the actual test net, I don't see the need to sweat over it.

**Comment by s3bc40** [comment]
I may have found why we all have the same error of ownership.

It is because we have to switch/import the anvil wallet to Metamask (already imported in previous courses on Account2).

When we do the same as Patrick on localhost anvil chain, it works with the `TokenId=0` and the contract address of the deployed `BasicNft`

Still, the image won't show up, but I think it should be something related to the local network since the ipfs url works on my side. The important part is that there is a response. 

As @EngrPips said, don't sweat it, and you can follow along locally (since sepolia testnet I always overshot in gas) :smiley: 

![image](https://github.com/user-attachments/assets/ecef50dc-9876-48a2-bee1-2127e489aa6e)

EDIT : I see that I am a little behind with the doggie, but the issue might be related with the Mood NFT too :sweat_smile:

**Reply by EngrPips** [reply]
Gas fee on sepolia have been a nightmare lately.
